addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1453,6 +1453,9 @@ const ColorpickerUserValueWidget = SelectUserValueWidget.extend({
                 this.colorPreviewEl.style.backgroundImage = this._value;
             }
         }
+        if (this.colorPalette.setSelectedColor) {
+            this.colorPalette.setSelectedColor(this._ccValue, this._value, false);
+        }
```

Comment:
Why this? It's already done on `open` (it would be the right place here but was somehow needed on open for efficiency in a stable fix... to review in master... not yet reviewed :grimacing:)

Reply:
Because when the palette is already opened, upon picking a gradient and keeping the popup opened, we need its content to reflect the new DOM state: gradient preview in custom button, horizontal gradient scale, sliders... (without switching to the initial tab)

Reply:
Indeed, did not think about the colorpicker kept opened now. It is worth a comment in the code :wink: